runtime/autoload/man.vim

```diff
@@ -118,15 +111,11 @@ endfunction
 " Expects a string like 'access' or 'access(2)'.
 function s:parse_page_and_section(sect, str) abort
   try
-    let save_isk = &iskeyword
-    setlocal iskeyword-=(,)
-    let page = substitute(a:str, '(*\(\k\+\).*', '\1', '')
-    let sect = substitute(a:str, '\(\k\+\)(\([^()]*\)).*', '\2', '')
-    if sect == page || -1 == match(sect, '^[0-9 ]\+$')
+    let [page, sect] = matchlist(a:str, '\v\C([-.[:alnum:]_]+)%(\(([-.[:alnum:]_]+)\))?')[1:2]
+    if empty(sect)
       let sect = a:sect
     endif
   catch
-    let &l:iskeyword = save_isk
```

Comment:
In the previous version &l:iskeyword did not get restored if everything is fine, now it is not modified. Do you think it was intentional and &iskeyword should be set to something specific after this function exits?